lib/metadataTypes/Automation.js

```diff
@@ -1558,6 +1558,116 @@ class Automation extends MetadataType {
+        this.definition.keyField = 'key';
+        this.definition.bodyIteratorField = 'entry';
+        const automationLegacyKeys = await super.retrieveREST(
+            undefined,
+            `/legacy/v1/beta/bulk/automations/automation/definition/`
+        );
+        this.definition.keyField = keyBackup;
+        this.definition.bodyIteratorField = iteratorBackup;
+        return automationLegacyKeys;
+    }
+    /**
+     * A function to update automation email notifications
+     *
+     * @param {string} keys metadata keys
+     * @returns {Promise.<string[]>} keys of the automations where notifications were updated
+     */
+    static async updateNotifications(keys) {
+        const emailComplete = Array.isArray(Util.OPTIONS.emailComplete)
+            ? Util.OPTIONS.emailComplete.join(',')
+            : Util.OPTIONS.emailComplete;
+        const emailError = Array.isArray(Util.OPTIONS.emailError)
+            ? Util.OPTIONS.emailError.join(',')
+            : Util.OPTIONS.emailError;
+        const updatedKeys = [];
+        let notificationsResult;
+        const automationLegacyMapObj = await this.#getEncodedAutomationIDs(); // retrieve automation legacy keys to update notifications
+        for (const key of Object.keys(automationLegacyMapObj.metadata)) {
+            if (Array.isArray(keys) && keys.includes(key)) {
+                try {
+                    notificationsResult = await this.client.rest.get(
+                        '/legacy/v1/beta/automations/notifications/' +
+                            automationLegacyMapObj.metadata[key].id
+                    );
+                } catch (ex) {
+                    Util.logger.error(
+                        `Error retrieving notifications for automation '${key}': ${ex.message} (${ex.code}))`
+                    );
+                }
+                // if the parameters provided are the same as the email addresses/notes already in the notifications - skip this automation
+                if (
+                    notificationsResult.workers.find(
+                        (notification) =>
+                            (notification.notificationType === 'Complete' &&
+                                (emailComplete !== notification.definition ||
+                                    Util.OPTIONS.completionNote !== notification.body)) ||
+                            notificationsResult.workers.find(
+                                (notification) =>
+                                    notification.notificationType === 'Error' &&
+                                    (emailError !== notification.definition ||
+                                        Util.OPTIONS.errorNote !== notification.body)
+                            )
+                    )
+                ) {
+                    // create payload
+                    const notificationBody = {
+                        programId: automationLegacyMapObj.metadata[key].id,
+                        workers: [
```

Comment:
this does not consider that 
- the user could choose to only update one of the 4 fields (2 email fields, 2 note fields)
- the user could try to update the note for error OR completion but provide no email --- AND that no corresponding email was set before
-  the user might want to REMOVE completion or error notification completely
- ... likely more variants in the same way. basically, we should provide full flexibility here.

Reply:
Now it's possible to only update either of the notes (if there are email addresses), either completion email or error email or both. --clear option allows to remove all notes and emails, only notes, only either type of emails. 

If someone tries to set a note and there is no email address, the user is asked if they want to set email address